x/evm/keeper/msg_server.go

```diff
@@ -153,9 +163,64 @@ func (k *Keeper) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams)
 	}
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if err := k.updatePrecompiles(ctx, req.Params.EnabledPrecompiles); err != nil {
+		return nil, err
+	}
+
 	if err := k.SetParams(ctx, req.Params); err != nil {
 		return nil, err
 	}
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
+
+func (k *Keeper) updatePrecompiles(ctx sdk.Context, newEnabledPrecompiles []string) error {
```

Comment:
Should we return an error if the precompile contract isn't registered in the geth precompile modules (RegisteredModules())?

Also small optional thing: I think this could be moved to a separate file, just feels it could be separate from the msg_server